src/index.ts

```diff
@@ -67,20 +90,39 @@ export class SgidClient {
    * @param scopes Array or space-separated scopes, must include openid
    * @param nonce Specify null if no nonce
    * @param redirectUri The redirect URI used in the authorization request, defaults to the one registered with the client
+   * @param codeChallenge The code challenge from the code verifier used for PKCE enhancement
+   * @param codeChallengeMethod The code challenge method used to generate the code challenge from the code verifier, must be `S256`
```

Comment:
yeah so i'm unsure about this because i omitted the check in line with trying to keep the SDK 'dumb' so that we can make changes to the server at our own cadence.

would this be something we want to check?

Reply:
after discussion with Jie Hao, we've decided NOT to include the `codeChallengeMethod` as a parameter. as there are currently only 2 values `plain` and `S256`, and the PKCE RFC clearly highlights that `plain` is insecure, there is no reason to let developer shoot themselves in the foot
(i.e. if we ONLY let them choose `S256`, whats the point of the param? OR if we let them choose between `plain` and `S256` but if they choose `plain` then we throw an error, whats the point of letting them choose?)

from a UX perspective, this reduces the mental load as users do not have to think about what code challenge method to employ

for users that are familiar with the PKCE code challenge methods, we will document on the developer docs that we are ONLY using `S256` as default due to security issues with using `plain`